hubblestack_nova/firewall.py

```diff
@@ -120,15 +120,17 @@ def audit(data_list, tags, verbose=False):
             control_reason = tag_data.get('control', '')
             description = tag_data.get('description')
             if (tag, description, control_reason) not in control_reasons:
-                dict = {'description': description,
+                tag_dict = {'description': description,
                         'control': control_reason}
-                controlled.append({tag: dict})
+                controlled.append({tag: tag_dict})
                 control_reasons.add((tag, description, control_reason))
 
         ret['Controlled'] = controlled
         ret['Success'] = success
         ret['Failure'] = failure
 
+    if len(ret['Controlled']) == 0: ret.pop('Controlled')
```

Comment:
Interestingly, an empty list in python is considered False. So instead you can just do

```
if not ret['Controlled']:
    ret.pop('Controlled')
```

Which is a little cleaner, I think.